src/kvstore/raftex/RaftPart.cpp

```diff
@@ -1162,7 +1167,14 @@ bool RaftPart::leaderElection() {
                 }
             }
             weight_ = 1;
-            sendHeartbeat();
+            sendHeartbeat().then([this] (folly::Try<AppendLogResult>&& t) {
+                CHECK(!t.hasException());
+                auto result = std::move(t).value();
+                if (result == AppendLogResult::SUCCEEDED) {
+                    std::lock_guard<std::mutex> g(raftLock_);
```

Comment:
Do we really need the lock here?

Reply:
You are right, I suppose `atomic_bool` would be enough.